### ISSUE TITLE
Support termination of child instances

### ISSUE THREAD

**Issue by remcowesterhoud** [camunda/camunda#9795]
# Description
We should be able to terminate instances called by another process. When this happens the child instance should be terminated. On the parent process an incident should be created, alerting users that the process is "stuck". This enables users to use instance modification to fix the parent process, terminate the parent manually, or retry the call activity.

- The child instance will be terminated
- An incident is created in the parent process, on the scope of the call activity
    - The message should indicate that the child has been terminated and that this process instance is stuck.
- Resolving this incident should retry the call activity.
    -  If the definition of the called process has been deleted, a new incident will be created that the called process does not exist (the same behaviour as usual)
    - We should re-apply the input mapping
    - We should reevaluate the processId

Blocked by: #9797

Note: This is specifically for the deletion of resources. This is also a valid case for the cancelling of a regular child instance. It would be neat to align regular termination with this so it behaves in the same way of the future.

Out of scope:
- Supporting termination of child instances outside of resource deletion

**Comment by saig0** [comment]
We might want to align the incident behavior with https://github.com/camunda/zeebe/issues/9704.

**Comment by megglos** [comment]
@saig0 I'm wondering about the use and need to create an incident in such case or if we could just mark the call activity instance as cancelled in that case? Similar to like this example where I cancelled a service task 
![image](https://github.com/camunda/zeebe/assets/209518/c067d4c2-795f-4180-a3d7-0d9281b132db)

The parent process might be stuck there but a user could decide on creating a new instance of the same activity or one further down if they want. wdyt?

**Comment by megglos** [comment]
Reasons for an incident:
* terminating a child could also terminate the root instance accidentally => thus raising an incident would prevent that and give the user the control to e.g. modify the instance etc., this is considered the better UX

**Comment by megglos** [comment]
rermoving from upcoming as this is planned via [this epic](https://github.com/camunda/product-hub/issues/2218) in the product hub
